code/roles/TTTRole.cs

```diff
@@ -50,7 +63,18 @@ public virtual void OnDeselect(TTTPlayer player)
 
         }
 
-        public virtual bool CanBuy() => false;
+        // serverside function
```

Comment:
Could put an `Assert(Host.IsServer)` if you want to make sure whenever someone calls this it's on the server.

Reply:
Could you provide an example?

Reply:
to the first line of `InitShop()` you can write...

`Assert(Host.IsServer)` and it'll throw an exception if `IsServer` is false.

Reply:
Did you tried this? As `Assert` is no function available in `Sandbox` namespace...

Reply:
`Host.AssertServer()` mb